ConduitIO/conduit
# deps(ember): upgrade to latest LTS (4.8.x)

## Description of change


Closes https://github.com/ConduitIO/conduit/issues/1005
Part of https://github.com/meroxa/product/issues/749

With this change, we're upgrading `conduit/ui` from `ember@4.0.1` to the latest LTS `ember@4.8.x`. This way, `conduit/ui` can also upgrade to [latest security patches provided to Ember](https://emberjs.com/releases/lts/) until December 2023.

This upgrade includes several minor version bumps in order to benefit from [a bug fix in @ember/data that just landed in the 4.9 release](https://github.com/emberjs/data/pull/8307). The related bug prevented us from upgrading beyond `ember@4.7` due to an error in identifying new data records wrapped with ember-changeset correctly (`Assertion Failed: Expected to receive a stable Identifier to subscribe to`).

In addition, this upgrade includes:

- pinned version definition for several dependencies to resolve a [sub dependency mismatch between @embroider/macros and ember-cli-mirage](https://github.com/miragejs/ember-cli-mirage/issues/2461)
- explicit enablement of Ember prototype extensions, which have been [deprecated recently](https://rfcs.emberjs.com/id/0848-deprecate-array-prototype-extensions/), but which are still in our codebase for now

## Resolutions

The following changes were made to upgrade:

- run `ember-cli-update --to 4.8`
- re-run `rm -rf node_modules && yarn` to remove the existing npm packages and start a fresh install
- run `ember t -s` to start the test suite and see errors related to the upgrade
- work through all of the errors mentioned below and re-build / re-run the test suite after each resolution to verify
- once tests pass, shut down the test server and re-run `sudo yarn start` to verify the changes locally at `https://dev.meroxa.io`

### Build error `@babel/template placeholder "SAMPLE_FEATURE_FLAG": Expected string substitution`

- you should see the `@embroider/macros` error in your terminal window (Build failure)
- open the log file mentioned in the error output
--> you should see the error stack trace pointing to the `node_modules` and you should see a reference to `ember-cli-mirage`
- pin resolutions for the following packages as follows in `package.json`:
```json
"resolutions": {
 // ...
     "miragejs": "~0.1.47",
    "@embroider/macros": "^1.10.0",
    "@embroider/util": "^1.10.0",
    "ember-cli-babel": "^7.26.11",
    "ember-modifier": "^3.2.7"
 // ...
 }
```
- re-run `rm -rf node_modules && yarn` to remove the existing npm packages and start a fresh install

## Blockers before merging

## How to verify?

+ checkout the feature branch and run the app locally
+ browse around
+ -> the app should work as expected and no errors should be thrown

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.
